GTG/core/task.py

```diff
@@ -344,53 +350,90 @@ def recursive_sync(self):
     def set_recurring(self, recurring: bool, recurring_term: str=None, newtask=False):
         """Sets a task as recurring or not, and its recurring term.
         
+        There are 4 cases to acknowledge when setting a task to recurring:
+            - if repeating but the term is invalid: it will be set to False.
+            - if repeating and the term is valid: we set it to True.
+            - if not repeating and the term is valid: we set the bool attr to True and set the term.
+            - if not repeating and the term is invalid: we set it to False and keep the previous term.
+
         Setting a task as recurrent implies that the
         children of a recurrent task will be also
         set to recurrent and will inherit 
         their parent's recurring term
         
         Args:
             recurring (bool): True if the task is recurring and False if not.
-            recurring_term (str, optional): the recurring period of a task (every Monday, day..). Defaults to None.
+            recurring_term (str, optional): the recurring period of a task (every Monday, day..).
+                                            Defaults to None.
             newtask (bool, optional): if this is a new task, we must set the due_date. Defaults to False.
-
-        Raises:
-            ValueError: if invalid recurring term
         """
-        # setting the task to recurrent
+        def is_valid_term():
+            """ Verify if the term is valid and returns the appropriate Due date.
+
+            Return a tuple of (bool, Date)
+            """
+            if recurring_term is None:
+                return False, None
+
+            try:
+                # If a start date is already set,
+                # we should calculate the next date from that day.
+                if self.start_date == Date.no_date():
+                    start_from = Date(convert_datetime_to_date(date.today()))
+                else:
+                    start_from = self.start_date
+
+                newdate = start_from.parse_from_date(recurring_term, newtask)
+                return (True, newdate)
+            except Exception as e:
```

Comment:
Do we know which exception might be thrown here?

Reply:
Yeah, that would be a `ValueError` from the raised from the date object when parsing the date in the a String format.
Should I explicitly put it in the exception?

Reply:
Yeah, it would be better to catch a explicit exception